### PR TITLE
chore: fix broken build setup

### PR DESCRIPTION
**Pull request by mjsax**
Upstream ZK version bump, requires to add new dependency.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -146,6 +146,7 @@
         <netty.version>4.1.100.Final</netty.version>
         <netty-codec-http2-version>4.1.100.Final</netty-codec-http2-version>
         <jersey-common>2.34</jersey-common>
+        <dropwizard.version>4.1.12.1</dropwizard.version>
```
Comment (lihaosky) [inline pom.xml:149]: Curious any reason we pick this version? This version is from 2020: https://mvnrepository.com/artifact/io.dropwizard.metrics/metrics-core/4.1.12.1

**Review by lihaosky — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (mjsax) [reply pom.xml:149]: It's the version used by ZK 3.8 -> https://github.com/apache/zookeeper/blob/branch-3.8/pom.xml#L478

**Comment by mjsax** [comment]
Build in 6.1.x and 6.2.x time out: https://jenkins.confluent.io/job/confluentinc/job/ksql/

Reason is that brokers don't startup, and thus clients loops forever with a connection error. -- Adding this missing dependency allows the server to startup, fixing the issue.

This fix is for 6.1.x and 6.2.x only. -- To avoid `pint-merge` issue, I did a 6.2.x PR in parallel so we can do a merge with `ours` without issues:

6.2.x PR: https://github.com/confluentinc/ksql/pull/10137

**Comment by mjsax** [comment]
Test `SystemAuthenticationFunctionalTest` failed -- we know that is needs a fix. Will merge this as-is, and we can tackle the test in a follow up PR.

**Comment by mjsax** [comment]
Merged with pint-merge strategy `ours`: https://jenkins.confluent.io/job/tools-autotasks/job/tools-autotasks/job/apply-pint-merge/3750/

**Pull request merged**